rust/rope/src/diff.rs

```diff
@@ -74,10 +87,13 @@ impl Diff<RopeInfo> for LineHashDiff {
         let mut prev_base = 0;
 
         let mut needs_subseq = false;
-        for line in target.lines_raw(start_offset..target_end) {
+        for line in SliceIter::new(base, start_offset) {
+            let len = line.range.end - line.range.start;
             let non_ws = non_ws_offset(&line);
-            if line.len() - non_ws >= MIN_SIZE {
-                if let Some(base_off) = line_hashes.get(&line[non_ws..]) {
+            if len - non_ws >= MIN_SIZE {
+                if let Some(base_off) = line_hashes
+                    .get(&RopeSlice { rope: base, range: line.range.end + non_ws..line.range.end })
```

Comment:
this should be `line.range.start+non_ws`? When I make this change though I notice that the deltas that are generated are different from what were previously generated, and it looks like there are a lot of empty `Copy` operations getting included; something else to investigate.